modules/light-clients/08-wasm/types/contract_api.go

```diff
@@ -96,55 +96,35 @@ type verifyUpgradeAndUpdateStateMsg struct {
 type checkSubstituteAndUpdateStateMsg struct{}
 
 // ContractResult defines the expected interface a Result returned by a contract call is expected to implement.
-type ContractResult interface {
-	Validate() bool
-	Error() string
-}
+type ContractResult interface{}
 
 // contractResult is the default implementation of the ContractResult interface and the default return type of any contract call
 // that does not require a custom return type.
-type contractResult struct {
-	IsValid  bool   `json:"is_valid,omitempty"`
-	ErrorMsg string `json:"error_msg,omitempty"`
-	Data     []byte `json:"data,omitempty"`
-}
-
-func (r contractResult) Validate() bool {
-	return r.IsValid
-}
-
-func (r contractResult) Error() string {
-	return r.ErrorMsg
-}
+type contractResult struct{}
 
 // statusResult is the expected return type of the statusMsg query. It returns the status of the wasm client.
 type statusResult struct {
-	contractResult
 	Status exported.Status `json:"status"`
 }
 
 // exportMetadataResult is the expected return type of the exportMetadataMsg query. It returns the exported metadata of the wasm client.
 type exportMetadataResult struct {
-	contractResult
 	GenesisMetadata []clienttypes.GenesisMetadata `json:"genesis_metadata,omitempty"`
 }
 
 // timestampAtHeightResult is the expected return type of the timestampAtHeightMsg query. It returns the timestamp for a light client
 // at a given height.
 type timestampAtHeightResult struct {
-	contractResult
-	Timestamp uint64 `json:"timestamp"`
+	Timestamp uint64 `json:"timestamp,omitempty"`
 }
 
 // checkForMisbehaviourResult is the expected return type of the checkForMisbehaviourMsg query. It returns a boolean indicating
 // if misbehaviour was detected.
 type checkForMisbehaviourResult struct {
-	contractResult
-	FoundMisbehaviour bool `json:"found_misbehaviour"`
+	FoundMisbehaviour bool `json:"found_misbehaviour,omitempty"`
```

Comment:
I think we remove the `omitempty`? I'm not sure if it is considered in the unmarshalling step.

Reply:
I thought we were missing them because of a mistake. Why would we not need them here?

Reply:
because we only umarshal these structs, right? I don't believe `omitempty` comes into play in the decoding stage.